### ISSUE TITLE
Sound error when moving the camera

### ISSUE THREAD

**Issue by Zylann** [godotengine/godot#4195]
Windows 7 64 bits, Godot 2.0.1

When the camera moves in _some specific ways_, an error occurs in the console:

```
ERROR: update: Invalid mix rate for voice (0) check for invalid pitch_scale param.
   At: servers/spatial_sound_2d/spatial_sound_2d_server_sw.cpp:1003
```

Then the voice that was playing cannot be controlled or stopped by code, and seems to loop forever if it was set to loop.

It happens often and randomly in my game and confuses sound feedbacks, so it took me hours to figure out how to reproduce this clearly in a minimal project:
http://zylannprods.fr/dl/godot/AudioError.zip
Just play the scene. A looping sound will play, then click to see the error happen.

**Comment by reduz** [comment]
this will be handled in 3.0

**Comment by Zylann** [comment]
I still get this error in 2.1. I know it's going to be fixed in 3.0 but now I found a more annoying side-effect of it, some sounds are not playing anymore in my game, as if the audio engine could not create any voice. I assume it happens rarely because that's the first time I have this effect, but just wanted to take note of it.

![invalid_mix_rate](https://cloud.githubusercontent.com/assets/1311555/20030912/c6cf9156-a36e-11e6-802a-974ca20bda60.png)

**Comment by reduz** [comment]
sound engine was completely rewritten, so closing this
